client-js/app/scripts/rpc.ts

```diff
@@ -0,0 +1,94 @@
+import promise = require('es6-promise');
+
+var Promise = promise.Promise;
+
+
+export class JsonRpcSocket {
+  url: string;
+  calls: { [id: number]: { reject: (any) => void; resolve: (error?: any) => void } };
+  nextId: number;
+  socket: WebSocket;
+  prefix: string;
+
+  openPromise: Promise<void>;
+  // A promise that will be resolved as soon as the socket is connected.
+  // We chain call promises off this promise so that the user can immediately
+  // make calls but we don't actually send any messages to the server until the
+  // socket is ready.
+  sendRequest(method: string, params: Object, socket, server: string): Promise<string> {
```

Comment:
We should change this to not take the final `server` parameter, since it's the caller's responsibility to construct the full method string. (registry service has been updated to reflect the change of responsibility, but this method and the echo service have not)